src/os/vxworks/inc/os-impl-tasks.h

```diff
@@ -35,7 +35,15 @@
 /*tasks */
 typedef struct
 {
+#if defined(OSAL_VXWORKS7)
+    /* SDS: WIND_TCB was moved to a private taskLib file. There is an API to access fields within. In this case,
+     * OSAL wants to store other data with WIND_TCB in a single allocaiton of memory, so we only need the size. */
+    char tcb[VX_WIND_TCB_SIZE]; /* Must be first */
+#elif defined(OSAL_VXWORKS6)
     WIND_TCB tcb;           /* Must be first */
+#else
+#error build system must define OSAL_VXWORKS6 or OSAL_VXWORKS7
+#endif
```

Comment:
Not sure why this was done with a separate macro (OSAL_VXWORKS6/7) vs. checking the macro that VxWorks 7 directly provides?  I'm also not fond of `#ifdef` directly inside struct declarations.

Since going forward future versions of VxWorks are likely to only provide VX_WIND_TCB_SIZE, then what I would prefer is something like:

```
#ifdef VX_WIND_TCB_SIZE
/* VxWorks >= 7.0 should provide this symbol via taskLib.h, WIND_TCB is an abstract type */
typedef char OS_VxWorks_TCB_t[VX_WIND_TCB_SIZE];
#else
/* Older VxWorks provide the definition of WIND_TCB directly */
typedef WIND_TCB OS_VxWorks_TCB_t;
#endif
```

Then the struct itself is simple - use `OS_VxWorks_TCB_t` as the type for the `tcb` member.

NOTE - when building coverage test the CMakeLists.txt inside vxworks dir isn't used - hence why this failed travis CI (neither is defined).  With the approach above it should work for both FSW and UT.

Reply:
Did you really just say you aren't fond of #ifdef directly inside struct declarations, and then suggest an #ifdef in a struct declaration? :)

Reply:
I like your idea better so I will remove the macros. If vxworks8 does it differently we will have to re-visit. :)

Reply:
Is "abstract type" the correct term to use in your comment? I think it should be "opaque type"

Reply:
No, the `#ifdef` is not _inside_ the struct declaration in my suggestion - it controls a typedef.  The typedef is then used in the struct - this is OK.  

This pattern means that the member always must exist and always have the same name - it might differ only in its type.  Putting `#ifdef` directly inside the struct means that names and even the existence of the member is uncertain - this is what I don't like about it. 

And yes, the term "opaque" is probably better here ... I do tend to intermix these terms.  C doesn't really do abstract types like C++ does.

Reply:
Sorry I missed the "typedef" I will change it now and push again.